kyaml/kio/pkgio_reader.go

```diff
@@ -285,8 +285,8 @@ func (r *LocalPackageReader) initReaderAnnotations(path string, _ os.FileInfo) {
 	}
 }
 
-// ShouldSkipDir returns a filepath.SkipDir if the directory should be skipped
-func (r *LocalPackageReader) ShouldSkipDir(path string, matcher *IgnoreFilesMatcher) error {
```

Comment:
Should we get rid of IncludeSubpackages flag as a whole in line 309 and reimplement the equivalent logic ? It's not clear as to how does this fit with ignoreKrmFiles

Reply:
I think there is an argument that the `LocalPackageReader` should not have the `IncludeSubPackages` flag, but we should instead read each subpackage separately. It fits better at least with the setter functionality. It seems unrelated to the ignoreFilesMatcher.